lms/services/application_instance.py

```diff
@@ -116,6 +116,19 @@ def search(
 
         return query.all()
 
+    def set_canvas_api_details(
+        self, application_instance, developer_key, developer_secret
+    ):
+
+        # If either one of developer_key or developer_secret is missing, then we won't set either
+        if developer_secret and developer_key:
```

Comment:
No need for the else side of this. If both are not present we just leave the AI untouched.

For new ones this would be `None`.